File: 0724-find-pivot-index/0724-find-pivot-index.c
int pivotIndex(int* nums, int numsSize) {
    
    int sum = 0;
    int leftSum = 0;
    for(int i = 0; i < numsSize; i++) {
        sum += nums[i];
    }

    for(int i = 0; i < numsSize; i++) {
        int rightSum = sum - leftSum - nums[i];
        if(leftSum == rightSum) return i;
        leftSum += nums[i];
    }

    return -1;
}

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution(object):
    def pivotIndex(self, nums):
        prefix_sum, suffix_sum = 0, sum(nums)
        n = len(nums)

        for i in range(n):
            suffix_sum -= nums[i]
            if suffix_sum == prefix_sum:
                return i
            prefix_sum += nums[i]
        
        return -1